data/lm/generate_package.py

```diff
@@ -124,24 +94,18 @@ def main():
     )
     parser.add_argument(
         "--force_utf8",
+        type=str,
         default="",
         help="Boolean flag, force set or unset UTF-8 mode in the scorer package. If not set, infers from the vocabulary. See <https://github.com/mozilla/DeepSpeech/blob/master/doc/Decoder.rst#utf-8-mode> for further explanation",
     )
     args = parser.parse_args()
 
-    if args.force_utf8 in ("True", "1", "true", "yes", "y"):
-        force_utf8 = Tristate(True)
-    elif args.force_utf8 in ("False", "0", "false", "no", "n"):
-        force_utf8 = Tristate(False)
-    else:
-        force_utf8 = Tristate(None)
-
```

Comment:
Before the flag check and the conversion was done here, but i moved it into the method to skip the conversion:

```
    if force_utf8 in ("True", "1", "true", "yes", "y"):
        use_utf8 = True
    elif force_utf8 in ("False", "0", "false", "no", "n"):
        use_utf8 = False
    else:
        use_utf8 = vocab_looks_char_based
        print("Using detected UTF-8 mode: {}".format(use_utf8))
```

This was the only usage of the Tristate class before.
And `use_utf8` in the next comparison is either True or False.

This is why i dont understand why we still need Tristate.

Reply:
I'm opposing this change not because it creates a bug or breaks functionality, but because it changes the code around for no good reason. Now instead of keeping flag parsing next to flag definition, it moves it into consumers, making encapsulation worse. If I decide to add a new consumer in the future I'd have to undo this change, or duplicate the parsing logic everywhere. With the Tristate class I have a consistent abstraction that I can pass around and avoids type confusion traps. It's unfortunate that Python doesn't have a better solution for this, but it's the price you pay in dynamic languages.

Reply:
The encapsulation is a good point.

What about the latest commit? This allows usage of True/False/None as method argument. 
I still would like to find a way without Tristate, to improve code readability.

Reply:
What exactly hurts readability about the Tristate class? Using plain booleans does not fix the problem that I have described again and again of implicitly casting to bool.

Reply:
It's an full class only needed to separate True/False/None.
I now changed the default case so you can insert any value into the method and it's using the detected mode if you did insert something other than True/False.

```
>>> f = False
>>> f is False or f is True
True
>>> f = True
>>> f is False or f is True
True
>>> f = None
>>> f is False or f is True
False
>>> f = 1
>>> f is False or f is True
False
```

Reply:
What about my last suggestion?
This allows you to insert any value (not just True/False/None) into the method and it's using the detected mode if you did insert something other than True/False.

If you still don't like it, I will add the Tristate class again.

Reply:
Please keep the Tristate class.

Reply:
Added again ...